File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/response/GetCurrentUserStatusResponse.java
package com.lattis.ellipse.data.network.model.response;

import com.google.gson.annotations.SerializedName;

/**
 * Created by ssd3 on 4/12/17.
 */

public class GetCurrentUserStatusResponse extends AbstractApiResponse{

    @SerializedName("payload")
    GetCurrentUserStatusPayloadResponse getCurrentUserStatusPayloadResponse;

    public GetCurrentUserActiveBookingStatusResponse getCurrentUserActiveBookingStatusResponse(){
        return  getCurrentUserStatusPayloadResponse.getCurrentUserActiveBookingStatusResponse;
    }

    public GetCurrentUserStatusTripResponse getCurrentUserStatusTripResponse(){
        return  getCurrentUserStatusPayloadResponse.getCurrentUserStatusTripResponse;
    }

    public String getSupportPhone(){
        return  getCurrentUserStatusPayloadResponse.getSupport_phone();
    }

    public String getOnCallOperator(){
        return  getCurrentUserStatusPayloadResponse.getOn_call_operator();
    }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/entity/response/user/current_status/GetCurrentUserStatusResponse.kt
package com.lattis.data.entity.response.user.current_status

import com.google.gson.annotations.SerializedName
import com.lattis.data.entity.response.base.AbstractApiResponse
import com.lattis.domain.models.DockHub

/**
 * Created by ssd3 on 4/12/17.
 */
class GetCurrentUserStatusResponse : AbstractApiResponse() {
    @SerializedName("payload")
    var getCurrentUserStatusPayloadResponse: GetCurrentUserStatusPayloadResponse? = null

    val currentUserActiveBookingStatusResponse: GetCurrentUserActiveBookingStatusResponse?
        get() = getCurrentUserStatusPayloadResponse?.getCurrentUserActiveBookingStatusResponse

    val currentUserStatusTripResponse: GetCurrentUserStatusTripResponse?
        get() = getCurrentUserStatusPayloadResponse?.getCurrentUserStatusTripResponse

    val supportPhone: String?
        get() = getCurrentUserStatusPayloadResponse?.support_phone

    val onCallOperator: String?
        get() = getCurrentUserStatusPayloadResponse?.on_call_operator

    val dockHub:DockHub?
    get() = getCurrentUserStatusPayloadResponse?.dockHub
}